healthtools/search/query.py

```diff
@@ -24,7 +26,12 @@ def run_query(query, doc_type=None):
     if not doc_type:
         return False, False
 
+    mc_key = remove_keywords(query + doc_type).replace(" ", "")
+    result = mc.get(mc_key)
+    if result:
+        return result, doc_type
```

Comment:
For the cache to be effective, it needs to be run before anything else and return essentially static content with minimal processing. This includes the doc_type.